File: src/Logs/telegram.ts
/**
 * Sends a message to a Telegram bot.
 * @param {string} log - The message to send to the Telegram bot.
 * @returns {Promise<void>} - A promise that resolves when the message is sent successfully, or rejects if there is an error.
 */
export const telegramBotChat = async (log: string): Promise<void> => {
  const tel_bot_token = process.env.TELEGRAM_BOT_TOKEN;
  const tel_chat_id = process.env.TELEGRAM_CHAT_ID;

  if (!tel_bot_token || !tel_chat_id) return;

  const controller = new AbortController();
  const timeout = setTimeout(() => controller.abort(), 10000);

  const tel_bot_api_route = `https://api.telegram.org/bot${tel_bot_token}/sendMessage`;
  const body = {
    chat_id: tel_chat_id,
    text: `*qBit-renamer*: \n${log}`,
    parse_mode: "Markdown",
  };

  try {
    await fetch(tel_bot_api_route, {
      signal: controller.signal,
      method: "POST",
      headers: {
        "content-type": "application/json; charset=utf-8",
      },
      body: JSON.stringify(body),
    });
  } catch (error) {
    console.error("Error sending message to Telegram bot");
    console.error(error);
    throw Error("Error sending message to Telegram bot");
  } finally {
    clearTimeout(timeout);
  }
};

File: Services/Telegram/telegram.go
package telegram

import (
	"bytes"
	"encoding/json"
	"fmt"
	config "helperr/Services/Config"
	"io"
	"net/http"
)

// telegramRequest sends a request to the Telegram Bot API.
// Parameters:
//   - method: HTTP method (GET, POST, etc.).
//   - url: API endpoint URL.
//   - body: Request body (io.Reader).
//
// Returns:
//   - []byte: Response body as byte slice.
//   - error: Any error encountered during the request.
func telegramRequest(method, url string, body io.Reader) ([]byte, error) {
	telCliente := &http.Client{}

	request, error := http.NewRequest(method, "https://api.telegram.org"+url, body)
	if error != nil {
		return nil, error
	}
	request.Header.Set("content-type", "application/json; charset=utf-8")

	response, error := telCliente.Do(request)
	if error != nil {
		return nil, error
	}

	responseBody, error := io.ReadAll(response.Body)
	if error != nil {
		return nil, error
	}
	defer response.Body.Close()

	return responseBody, nil
}

// TelegramBotChat sends a message to a Telegram bot.
// Parameters:
//   - log: The message to send.
//
// Returns:
//   - None
func TelegramBotChat(log string) {
	telBotTocken := config.Env.TelegramBotToken
	telChatId := config.Env.TelegramChatId

	if telBotTocken == "" || telChatId == 0 {
		return
	}

	body := map[string]interface{}{
		"chat_id":    telChatId,
		"text":       fmt.Sprintf("*Helperr*: \n %s", log),
		"parse_mode": "Markdown",
	}

	bodyJson, error := json.Marshal(body)
	if error != nil {
		println("Telegram | Error |", error.Error())
		return
	}

	url := fmt.Sprintf("/bot%s/sendMessage", telBotTocken)
	_, error = telegramRequest("POST", url, bytes.NewBuffer(bodyJson))
	if error != nil {
		println("Telegram | Error | error sending message to Telegram bot")
	}
}
